### PR TITLE
Master website sale slides improve course purchase jeh

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_product_configurator/static/src/js/website_sale_options.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ publicWidget.registry.WebsiteSale.include({
             },
             success: function (quantity) {
                 if (goToShop) {
-                    var path = window.location.pathname.replace(/shop([\/?].*)?$/, "shop/cart");
+                    var path = "/shop/cart";
```
Comment (tde-banana-odoo) [inline addons/website_sale_product_configurator/static/src/js/website_sale_options.js]: Why this change ?

File: addons/website_sale_slides/__manifest__.py [hunk 2]
```diff
@@ -9,7 +9,6 @@
 
     'depends': ['website_slides', 'website_sale'],
     'installable': True,
-    'auto_install': True,
```
Comment (tde-banana-odoo) [inline addons/website_sale_slides/__manifest__.py]: Is it intended ?

File: addons/website_sale_slides/data/website_sale_slides_demo.xml [hunk 3]
```diff
@@ -17,4 +17,42 @@
         <field name="product_id" ref="product_course"/>
     </record>
 
+    <!-- CHANNEL 1: Taking care of Trees -->
+    <!-- ================================================== -->
+    <record id="water_can" model="product.product">
```
Comment (tde-banana-odoo) [inline addons/website_sale_slides/data/website_sale_slides_demo.xml]: Small remark: I generally vote for neutral IDs for demo. That way if we change the content of the demo we don't have to change its xml ID . For example here I would have something like product_course_channel_1_option_0 so that we know this is the first option of the product_course linked to channel_1 .

File: addons/website_sale_slides/views/website_slides_templates.xml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" ?>
 <odoo><data>
 
-<template id='course_sidebar' inherit_id="website_slides.course_sidebar">
+<template name="Product" id='course_sidebar' inherit_id="website_slides.course_sidebar">
```
Comment (tde-banana-odoo) [inline addons/website_sale_slides/views/website_slides_templates.xml]: Good idea to add a name! I would suggest to improve it a bit because Product is maybe a bit generic: maybe something like "Join Channel: Product Button" or something like that.

File: addons/website_sale_slides/views/website_slides_templates.xml [hunk 5]
```diff
@@ -31,4 +42,16 @@
     </xpath>
 </template>
 
+<template id="buy_now_course"
```
Comment (tde-banana-odoo) [inline addons/website_sale_slides/views/website_slides_templates.xml]: Side note: indentation seems strange here. We could also update the id to something like course_sidebar_buy_now_course or something like that to be more specific.

File: addons/website_sale_product_configurator/static/src/js/website_sale_options.js (hunk 1 above)
Comment (jeh-odoo) [reply addons/website_sale_product_configurator/static/src/js/website_sale_options.js]: Before, this part was called when we were in the shop so with a URL containing "/shop/". But when we are in a course the URL is not "/shop/" but "/slides/". So when we try to buy a course with the button "Add to Cart" and we click on proceed checkout, we weren't redirect on the cart because of the regex.
I didn't find an utility to keep the regex while we are redirect to the cart anyway in both situation.

File: addons/website_sale_slides/__manifest__.py (hunk 2 above)
Comment (jeh-odoo) [reply addons/website_sale_slides/__manifest__.py]: I deleted this to test when we unchecked the option to sell courses if the buttons weren't present anymore. But when I try in the first place, I couldn't do it because each time I unchecked the option it was automatically reinstall just after.

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo rebase-ff
